contrib/test/integration/main.yml

```diff
@@ -1,7 +1,34 @@
-- hosts: all
-  remote_user: root
+---
+
+- hosts: '{{ subjects | default("all") }}'
+  gather_facts: False  # Requires low-level ansible-dependencies
+  vars_files:
+    - "{{ playbook_dir }}/vars.yml"
+  tags:
+    - setup
+  tasks:
+    - name: Ensure low-level ansible-dependencies are installed
+      raw: $(type -P dnf || type -P yum) install -y python2 python2-dnf libselinux-python git rsync
```

Comment:
This will fail on RHEL: there's no such package as `python2`.

Reply:
And yet it works perfectly...the Ansible ``raw`` module doesn't do any exit-status checking.  If something really went wrong here, the user will find out about it later when ansible pukes about something missing.

However, you're right to point this out in that it's a bit too much JFM.  Let me add a comment  about this, since detecting Fedora vs RHEL at this stage is not possible (setup hasn't run yet).